packages/patrol_cli/lib/src/coverage/coverage_tool.dart

```diff
@@ -0,0 +1,213 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:adb/adb.dart';
+import 'package:coverage/coverage.dart' as coverage;
+import 'package:dispose_scope/dispose_scope.dart';
+import 'package:file/file.dart';
+import 'package:glob/glob.dart';
+import 'package:patrol_cli/src/base/logger.dart';
+import 'package:patrol_cli/src/coverage/device_to_host_port_transformer.dart';
+import 'package:patrol_cli/src/coverage/vm_connection_details.dart';
+import 'package:patrol_cli/src/devices.dart';
+import 'package:process/process.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:vm_service/vm_service_io.dart';
+
+class CoverageTool {
+  CoverageTool({
+    required FileSystem fs,
+    required ProcessManager processManager,
+    required Adb adb,
+    required DisposeScope parentDisposeScope,
+  })  : _fs = fs,
+        _processManager = processManager,
+        _adb = adb,
+        _disposeScope = DisposeScope() {
+    _disposeScope.disposedBy(parentDisposeScope);
+  }
+
+  final FileSystem _fs;
+  final ProcessManager _processManager;
+  final Adb _adb;
+  final DisposeScope _disposeScope;
+
+  Future<void> run({
+    required String flutterPackageName,
+    required TargetPlatform platform,
+    required Logger logger,
+    required Set<Glob> ignoreGlobs,
+  }) async {
+    final homeDirectory =
+        Platform.environment['HOME'] ?? Platform.environment['USERPROFILE'];
```

Comment:
This is also a hidden dependency on `io.Platform` now. Use `Platform` from [`package:platform`](https://pub.dev/packages/platform) (just like is now done with `FileSystem` and `ProcessManager`)

Example: https://github.com/leancodepl/patrol/blob/370a0fe808074d7edfae5cbcc48249be01c16d47/packages/patrol_cli/lib/src/commands/doctor.dart#L13